wpilibc/src/main/native/include/frc/apriltag/AprilTagFieldLayout.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <string_view>
+#include <vector>
+
+#include <wpi/SymbolExports.h>
+
+#include "frc/DriverStation.h"
+#include "frc/apriltag/AprilTag.h"
+#include "frc/geometry/Pose3d.h"
+
+namespace wpi {
+class json;
+}  // namespace wpi
+
+namespace frc {
+class WPILIB_DLLEXPORT AprilTagFieldLayout {
+ public:
+  std::vector<AprilTag> m_apriltags;
```

Comment:
This should be private if possible.

Reply:
It is public because of the to_json method, which does not have access to private variables in AprilTagFieldLayout.

Reply:
In that case, declare from_json() and to_json() as friends.